source/server-discovery-and-monitoring/server-discovery-and-monitoring-tests.rst

```diff
@@ -209,3 +209,38 @@ This test requires MongoDB 4.9.0+.
 
```

Comment:
I think we're missing a test here that verifies the minPoolSize -> SDAM communication. 

What do you think about this test:
1. Create setup client with directConnection=true
2. Enable the following failpoint using the setup client: 
```
     {
         configureFailPoint: "failCommand",
         mode: { skip: 2 },
         data: {
             failCommands: ["isMaster"],
             errorCode: 1234,
             appName: "SDAMMinPoolSizeError"
         }
     }
```
3. Create client with directConnection=true, appName="SDAMMinPoolSizeError", minPoolSize=10, serverSelectionTimeoutMS=100, heartbeatFrequencyMS=10000

4. Verify via SDAM and CMAP event monitoring on the client that a ConnectionPoolReadyEvent occurs
   after the first ServerHeartbeatSucceededEvent event does.

5. Verify that an PoolClearedEvent is emitted

6. Execute a `ping` command using the client and assert that it fails due to server selection timeout

7. Disable the failpoint using the setup client

8. Verify via SDAM and CMAP event monitoring that a ConnectionPoolReadyEvent occurs
   after the next ServerHeartbeatSucceededEvent event.

9. Execute a `ping` command using the client and assert that it succeeds.

Reply:
I think this test can be implemented with the SDAM integration format because the assertion in step 4) isn't that critical to this test. The main point is that the background thread clears the pool, right?

Reply:
Yeah the assertion in step 4 isn't necessary to this test, and the one in step 8 is also not really necessary, though it would be nice to have. I went ahead and took a stab at writing an SDAM spec test, but Rust doesn't have the runner (nor SDAM monitoring) implemented, so I was driving a bit blind. Let me know if the test parses + passes in Python.